specs/state-proof.md

```diff
@@ -79,10 +83,12 @@ to not be in the table.
 - 6.2. `value` is 0 if first access and READ
 - 6.3. `initial value` is 0
 - 6.4. `state root` is the same
+- 6.5. `value_prev` is 0
 
 ### Account
 - 7.0. `id` and `storage_key` are 0
 - 7.1. MPT storage lookup for last access to (address, field_tag)
+- 7.2. `value_prev` equals `initial_value`
 
 ### Tx Refund
```

Comment:
8.1 to be changed to something like : `state_root` equals  `state_root_prev`

p.s. nitpicking: in the code this section is 7.x so needs to be updated

Reply:
Constraint in spec 8.3 and code differs:
`- 8.3. First access for a set of all keys are 0 if `READ``

```
        self.condition(q.not_first_access.clone(), |cb| {
            cb.require_word_equal(
                "value column at Rotation::prev() equals value_prev at Rotation::cur()",
                q.rw_table.value_prev.clone(),
                q.value_prev_column(),
            );
        });
```
From the general constraints, we have that at the first access, the value_prev_column equals the initial value. From 8.2 we have that init value = 0. And now we have that, at all but the first access, the value at the previous rotation equals value value_prev_column. So all values are indeed set at 0 but there is no if READ condition. Is that implicit?

edit: should that be a new constraint as in 9.5 actually and so 8.3 is missing?